Extensions/ExternalTfs/Src/Tasks/DownloadExternalBuildArtifacts/download.ts

```diff
@@ -55,27 +55,44 @@ function publishEvent(feature, properties: any): void {
 
 async function main(): Promise<void> {
     var promise = new Promise<void>(async (resolve, reject) => {
-        var projectId = tl.getInput("project", true);
-        var definitionId = tl.getInput("definition", true);
-        var buildId = tl.getInput("version", true);
-        var itemPattern = tl.getInput("itemPattern", true);
-        var downloadPath = tl.getInput("downloadPath", true);
-
-        var endpointUrl : any;
-        var username : any;
-        var accessToken : any;
-
-        if(isAdoServiceConnectionSet()) {
-            var connection = tl.getInput("azureDevOpsServiceConnection", true);
-            endpointUrl = tl.getEndpointUrl(connection, false);
+        // Read inputs from the user
+        var connectionType = tl.getInput('connectionType', false);
+        var connectionClassic = tl.getInput('connection', false);
+        var connectionAdo = tl.getInput('azureDevOpsServiceConnection', false);
+        var projectClassic = tl.getInput('project', false);
+        var projectAdo = tl.getInput('projectAdo', false);
+        var versionClassic = tl.getInput('version', false);
+        var versionAdo = tl.getInput('versionAdo', false);
+        var itemPattern = tl.getInput('itemPattern', false);
+        var downloadPath = tl.getInput('downloadPath', true);
+
+        // Determine inputs for fetching artifacts based on the selected connection type
+        var isAdoConnectionType = connectionType === 'ado';
+        var serviceConnection = isAdoConnectionType ? connectionAdo : connectionClassic;
+        var projectId = isAdoConnectionType ? projectAdo : projectClassic;
+        var buildId = isAdoConnectionType ? versionAdo : versionClassic;
+
+        if (!serviceConnection || !projectId || !buildId) {
+            const error = "Required inputs (service connection, project ID and build ID must be provided.";
```

Comment:
Ideally we should be more specific here to make user's life easier, so better to have 3 separate messages.